ios/RNDidomi.swift

```diff
@@ -381,60 +381,128 @@ class RNDidomi: RCTEventEmitter {
     }
 
     @objc(setUser:algorithm:secretId:salt:digest:resolve:reject:)
-    dynamic func setUser(id: String, algorithm: String?, secretId: String?, salt: String?, digest: String?, resolve:RCTPromiseResolveBlock, reject:RCTPromiseRejectBlock) {
-        if let algorithm = algorithm, let secretId = secretId, let digest = digest {
+    dynamic func setUser(id: String?, algorithm: String?, secretId: String?, salt: String?, digest: String?, resolve:RCTPromiseResolveBlock, reject:RCTPromiseRejectBlock) {
+        if let id = id, let algorithm = algorithm, let secretId = secretId, let digest = digest {
             Didomi.shared.setUser(id: id, algorithm: algorithm, secretId: secretId, salt: salt, digest: digest)
+        } else if let id = id {
+            DispatchQueue.main.async {
+                if let containerController = RCTPresentedViewController() {
+                    Didomi.shared.setUser(id: id, containerController: containerController)
+                } else {
+                    Didomi.shared.setUser(id: id)
+                }
+            }
         } else {
-            Didomi.shared.setUser(id: id)
+            Didomi.shared.clearUser()
         }
         resolve(0)
     }
 
     @objc(setUserWithHashAuth:algorithm:secretId:digest:salt:resolve:reject:)
     dynamic func setUserWithHashAuth(id: String, algorithm: String, secretId: String, digest: String, salt: String?, resolve:RCTPromiseResolveBlock, reject:RCTPromiseRejectBlock) {
-        Didomi.shared.setUser(
-            userAuthParams: UserAuthWithHashParams(
-                id: id,
-                algorithm: algorithm,
-                secretID: secretId,
-                digest: digest,
-                salt: salt))
+        DispatchQueue.main.async {
+            if let containerController = RCTPresentedViewController() {
+                    Didomi.shared.setUser(
```

Comment:
since the userAuthParams parameter is the same for both cases, maybe it can be created before the if statement